src/riscv-double-trap.adoc

```diff
@@ -137,12 +137,12 @@ An `SRET` instruction sets the `SDT` bit to 0.
 
 [NOTE]
 ====
-A trap handler after saving the state needed for resuming from the trap,
-including `scause`, `sepc`, and `stval` among others, should clear the `SDT` bit
-when it is reentrant.
+After a trap handler, such as `scause`, `sepc`, and `stval` 
+among others, saves the state that is needed to resume from the trap, it
+should clear the `SDT` bit when it is reentrant.
 
-Resetting of the `SDT` by an `SRET` enables the trap handler to detect double
-trap occuring during the tail phase, where it restores critical state to return
+If the `SDT` is reset by an `SRET`, the trap handler is enabled to detect double
+traps that occur during the tail phase, where it restores critical state to return
```

Comment:
This is explaining why the ISA decided to reset the `SDT`. I think original sentence is good.

Reply:
Starting a sentence with a gerund can be confusing for translation. This sentence is also passive so making it more active can help with understanding. Can we compromise with this? "Resetting the `SDT` by an `SRET` enables the trap handler to detect a double trap that occurs during the tail phase, where it restores a critical state to return..."?

Reply:
Yes. That would be good.